app/design-system/design-system-internals/src/main/kotlin/com/hedvig/android/design/system/internals/SnackBar.kt

```diff
@@ -0,0 +1,158 @@
+  icon: @Composable () -> Unit,
+  modifier: Modifier = Modifier,
+  action: (() -> Unit)? = null,
+  actionLabel: String? = null,
+) {
+  val snackbarHostState = remember { SnackbarHostState() }
+  LaunchedEffect(showSnackbar, snackbarText) {
+    if (!showSnackbar) return@LaunchedEffect
+    val result = snackbarHostState.showSnackbar(snackbarText, actionLabel = actionLabel)
+    when (result) {
+      Dismissed -> {
+        showedSnackbar()
+      }
+
+      ActionPerformed -> {
+        if (action != null) {
+          action()
+        }
+        showedSnackbar()
+      }
+    }
+  }
+  SnackbarHost(snackbarHostState, modifier.padding(16.dp)) { _ ->
+    Snackbar(
+      action = {
+        if (actionLabel != null) {
+          ActionContent(
+            actionLabel,
+            textStyle,
+            modifier = Modifier.clip(shape).clickable(enabled = true, onClick = action ?: showedSnackbar).padding(
+              top = 16.dp,
+              bottom = 16.dp,
+              start = 8.dp,
+              end = 8.dp,
+            ),
+          )
+        }
+      },
+      dismissAction = null,
+      actionOnNewLine = false,
+      shape = shape,
+      containerColor = colors.containerColor,
+      contentColor = colors.textColor,
+      actionContentColor = colors.textColor,
+    ) {
+      if (actionLabel != null) {
+        SnackContentForAction(
+          icon = icon,
+          snackbarText = snackbarText,
+          textStyle = textStyle,
+          modifier = Modifier.padding(
+            top = 16.dp,
+            bottom = 16.dp,
+            end = 8.dp,
+          ),
+        )
+      } else {
+        PlainSnackContent(
+          icon = icon,
+          snackbarText = snackbarText,
+          textStyle = textStyle,
+          modifier = Modifier
+            .padding(
+              top = 16.dp,
+              bottom = 16.dp,
+            )
+            .fillMaxWidth(),
+        )
+      }
+    }
+  }
+}
+
+@Composable
+private fun ActionContent(actionLabel: String, textStyle: TextStyle, modifier: Modifier) {
+  Text(
+    actionLabel,
+    style = textStyle.copy(textDecoration = TextDecoration.Underline),
+    modifier = modifier,
+  )
+}
+
+@Composable
+private fun PlainSnackContent(
+  icon: @Composable () -> Unit,
+  snackbarText: String,
+  textStyle: TextStyle,
+  modifier: Modifier,
+) {
+  Row(
+    horizontalArrangement = Arrangement.Center,
+    verticalAlignment = Alignment.CenterVertically,
+    modifier = modifier,
+  ) {
+    icon()
+    Spacer(Modifier.width(8.dp))
+    Text(snackbarText, style = textStyle)
+  }
+}
+
+@Composable
+private fun SnackContentForAction(
+  icon: @Composable () -> Unit,
+  snackbarText: String,
+  textStyle: TextStyle,
+  modifier: Modifier,
+) {
+  Row(
+    horizontalArrangement = Arrangement.Start,
+    verticalAlignment = Alignment.CenterVertically,
+    modifier = modifier,
+  ) {
+    icon()
+    Spacer(Modifier.width(8.dp))
+    Text(snackbarText, style = textStyle)
+  }
+}
```

Comment:
These two seem to be identical, is it on purpose?

Reply:
The horizontalArrangement is different, but you're right, it's better be one function:)